bool done
int16 withX
int16 withY